### PR TITLE
Update components radius

### PR DESCRIPTION
**Pull request by jameskoster**
Most of the changes here do not lead to any visual differences. Those that do are a consequence of adopting the expanded radius scale and noted below. As a reminder here's the current scale and guidelines:

- `$radius-x-small:` Applied to elements like buttons nested within primitives like inputs.
- `$radius-small`: Applied to most primitives.
- `$radius-medium`: Applied to containers with smaller padding.
- `$radius-large`: Applied to containers with larger padding.
- `$radius-full`: For pills.
- `$radius-round`: For circles and ovals.

## Color palette

The color card in the picker has increased radius. 

| Before | After |
| --- | --- |
| Screenshot 2024-08-08 at 16 28 52 | Screenshot 2024-08-08 at 16 28 37 |

## Guide

The modal has increased radius. Arguably this should be inherited from the main `Modal` component—how do y'all feel about making that change here?

| Before | After |
| --- | --- |
| Screenshot 2024-08-08 at 16 32 30 | Screenshot 2024-08-08 at 16 32 39 |

## Modal

Increased radius on outer container.

| Before | After |
| --- | --- |
| Screenshot 2024-08-08 at 16 34 11 | Screenshot 2024-08-08 at 16 34 22 |

## Placeholder

Increased radius on outer container.

> [!NOTE]  
> This change is only visible in Storybook. It seems that _block_ placeholders override this, so there will be no visual change in-product.

| Before | After |
| --- | --- |
| Screenshot 2024-08-08 at 16 36 44 | Screenshot 2024-08-08 at 16 36 52 |

## Popover

Increased radius on outer container.

| Before | After |
| --- | --- |
| Screenshot 2024-08-08 at 16 38 52 | Screenshot 2024-08-08 at 16 39 00 |

## Snackbar

Increased radius on outer container.

| Before | After |
| --- | --- |
|  Screenshot 2024-08-08 at 16 40 19 | Screenshot 2024-08-08 at 16 40 26 |

## Toggle Group Control

Decreased radius on inner buttons.

| Before | After |
| --- | --- |
| Screenshot 2024-08-08 at 16 42 06 | Screenshot 2024-08-08 at 16 42 19 |

## Unit control

Decreased radius on inner button.

| Before | After |
| --- | --- |
| Screenshot 2024-08-08 at 16 45 03 | Screenshot 2024-08-08 at 16 45 11 |

### CODE REVIEW

File: packages/base-styles/_variables.scss [hunk 1]
```diff
@@ -47,7 +47,7 @@ $radius-x-small: 1px;   // Applied to elements like buttons nested within primit
 $radius-small: 2px;     // Applied to most primitives.
 $radius-medium: 4px;    // Applied to containers with smaller padding.
 $radius-large: 8px;     // Applied to containers with larger padding.
-$radius-full: 9999px;   // For lozenges.
+$radius-full: 9999px;   // For pills.
```
Comment (tyxla) [inline packages/base-styles/_variables.scss:50]: 👍 Thank you for this. As a non-native speaker, the difference between tablet, pill, lozenge, pellet, capsule, etc. makes my head hurt. `pill` does like a word that's more common than `lozenge`.

File: packages/components/src/progress-bar/styles.ts [hunk 2]
```diff
@@ -40,7 +40,7 @@ export const Track = styled.div`
 		${ COLORS.theme.foreground },
 		transparent 90%
 	);
-	border-radius: ${ CONFIG.radiusBlockUi };
+	border-radius: ${ CONFIG.radiusFull };
```
Comment (tyxla) [inline packages/components/src/progress-bar/styles.ts:43]: This used to be 2px and is now 9999px. I also don't see this in the list of changes in the PR description. Did you mean to use `radiusSmall` or `radiusMedium`?

File: packages/components/src/progress-bar/styles.ts [hunk 3]
```diff
@@ -58,7 +58,7 @@ export const Indicator = styled.div< {
 	position: absolute;
 	top: 0;
 	height: 100%;
-	border-radius: ${ CONFIG.radiusBlockUi };
+	border-radius: ${ CONFIG.radiusFull };
```
Comment (tyxla) [inline packages/components/src/progress-bar/styles.ts:61]: Same as above.

File: packages/components/src/guide/style.scss [hunk 4]
```diff
@@ -9,7 +9,6 @@
 	.components-modal__content {
 		padding: 0;
 		margin-top: 0;
-		border-radius: $radius-block-ui;
```
Comment (tyxla) [inline packages/components/src/guide/style.scss:12]: Agree to keep this consistent with the dialog/modal component we use 👍

File: packages/components/src/alignment-matrix-control/styles/alignment-matrix-control-styles.ts [hunk 5]
```diff
@@ -7,15 +7,15 @@ import { css } from '@emotion/react';
...
+import { COLORS, CONFIG } from '../../utils';
 import type {
 	AlignmentMatrixControlProps,
 	AlignmentMatrixControlCellProps,
 } from '../types';
 
 export const rootBase = () => {
 	return css`
-		border-radius: 2px;
+		border-radius: ${ CONFIG.radiusMedium };
```
Comment (tyxla) [inline packages/components/src/alignment-matrix-control/styles/alignment-matrix-control-styles.ts:18]: Did you mean to increase this to `4px`? I don't see it in the list of intended changes.

File: packages/components/src/toolbar/toolbar-group/style.scss [hunk 6]
```diff
@@ -1,7 +1,6 @@
 .components-toolbar-group {
 	min-height: $block-toolbar-height;
 	border-right: $border-width solid $gray-900;
-	background-color: $white;
```
Comment (tyxla) [inline packages/components/src/toolbar/toolbar-group/style.scss]: This feels like something unintended. Should it be part of another PR? I know we're overriding the background color here and there, so we might want to audit those instances separately, for example:

https://github.com/WordPress/gutenberg/blob/base-control-margin/packages/block-editor/src/components/block-toolbar/style.scss#L26

**Review by tyxla — COMMENTED** [review]
Thanks for working on it @jameskoster 🙌 

This mostly looks good in my testing ✅  

I've asked a few questions about some of the changes - let me know what you think.

Looks like you'll need to update a bunch of test snapshots. Happy to help with that once you confirm all changes are expected. 

Finally, I'd definitely add changelog entries here. Perhaps it makes sense to list all affected components.

File: packages/components/src/alignment-matrix-control/styles/alignment-matrix-control-styles.ts (hunk 5 above)
Comment (jameskoster) [reply packages/components/src/alignment-matrix-control/styles/alignment-matrix-control-styles.ts:18]: My feeling is this falls into 'small container' territory. That said, I don't think the radius of this component is visually rendered anywhere?

File: packages/components/src/progress-bar/styles.ts (hunk 2 above)
Comment (jameskoster) [reply packages/components/src/progress-bar/styles.ts:43]: Good question. Visually there's no change here which is why I didn't list it. 

The design of the progress bar calls for a fully rounded track/indicator. Since it is only 2px tall `radiusXSmall` (`2px`) is _technically_ adequate. However if we ever decide to increase the height of the progress bar then the track/indicator would no longer be fully rounded. But utilising `radiusFull` we ensure the bar is fully rounded regardless of how tall it is.

File: packages/components/src/toolbar/toolbar-group/style.scss (hunk 6 above)
Comment (jameskoster) [reply packages/components/src/toolbar/toolbar-group/style.scss]: The change is intended; without it, the group background obscures the radius of the container:

Screenshot 2024-08-12 at 20 21 47

We could add a radius to the group instead of removing the background, but that causes issues with the separator styling:

Screenshot 2024-08-12 at 20 22 45

I don't mind extracting the toolbar change from this PR and updating it separately though. In fact that's probably a good idea given it's likely the most impactful change. We can discuss the intricacies in a different PR.

File: packages/components/CHANGELOG.md [hunk 7]
```diff
@@ -20,6 +20,7 @@
 -   `SelectControl`: Infer `value` type from `options` ([#64069](https://github.com/WordPress/gutenberg/pull/64069)).
 -   `Guide`: Add `__next40pxDefaultSize` to buttons ([#64181](https://github.com/WordPress/gutenberg/pull/64181)).
 -   `SelectControl`: Pass through `options` props ([#64211](https://github.com/WordPress/gutenberg/pull/64211)).
+-   `AlignmentMatrixControl`, `AnglePickerControl`, `BorderControl`, `ButtonGroup`, `Button`, `CircularOptionPicker`, `ColorIndicator`, `ColorPalette`, `CustomGradientPicker`, `DropZone`, `DropdownMenuV2`, `FocalPointPicker`, `Guide`, `Modal`, `Placeholder`, `Popover`, `ProgressBar`, `RadioControl`, `Snackbar`, `TabPanel`, `Text`, `ToggleGroupControl`, `ToolbarGroup`, `Toolbar`, `Tooltip`, `UnitControl`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368)).
```
Comment (jameskoster) [inline packages/components/CHANGELOG.md]: Not sure how best to format this :)

Comment (tyxla) [reply packages/components/CHANGELOG.md]: A readable alternative is to just do separate entries, it's fine to have multiple entries per PR:

```suggestion
-   `AlignmentMatrixControl`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `AnglePickerControl`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `BorderControl`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `ButtonGroup`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `Button`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `CircularOptionPicker`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `ColorIndicator`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `ColorPalette`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `CustomGradientPicker`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `DropZone`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `DropdownMenuV2`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `FocalPointPicker`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `Guide`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `Modal`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `Placeholder`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `Popover`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `ProgressBar`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `RadioControl`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `Snackbar`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `TabPanel`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `Text`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `ToggleGroupControl`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `ToolbarGroup`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `Toolbar`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `Tooltip`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
-   `UnitControl`: Adopt radius scale. ([#64368](https://github.com/WordPress/gutenberg/pull/64368))
```

File: packages/components/src/progress-bar/styles.ts (hunk 2 above)
Comment (tyxla) [reply packages/components/src/progress-bar/styles.ts:43]: > Since it is only 2px tall radiusXSmall (2px) is technically adequate.

I assume you meant `radiusSmall` (2px) since `radiusXSmall` is `1px`. 

Anyway, that's not a problem for your intent here, and making it rounded makes sense to me 👍

File: packages/components/src/alignment-matrix-control/styles/alignment-matrix-control-styles.ts (hunk 5 above)
Comment (tyxla) [reply packages/components/src/alignment-matrix-control/styles/alignment-matrix-control-styles.ts:18]: That's correct. But why do we even need the `border-radius` then? It feels like it's redundant.

If it's there just for the case when someone provides a custom background color, then why can't the consumer also add a border-radius when they need it?

File: packages/components/src/toolbar/toolbar-group/style.scss (hunk 6 above)
Comment (tyxla) [reply packages/components/src/toolbar/toolbar-group/style.scss]: Yeah, I'd prefer a separate PR if that's possible. But I'll leave that call to you, it doesn't feel like this is a blocker for this PR anyway.

**Review by tyxla — APPROVED** [review]
LGTM 👍 Thanks @jameskoster 🙌 

Part of it can be split out to another PR, as discussed above, but that's not a blocker.

CHANGELOG will need a rebase before 🚢 .

File: packages/components/src/alignment-matrix-control/styles/alignment-matrix-control-styles.ts (hunk 5 above)
Comment (jameskoster) [reply packages/components/src/alignment-matrix-control/styles/alignment-matrix-control-styles.ts:18]: I agree, but it seems safer to adjust the value here rather than remove it? We could remove it in a follow up with a dedicated changelog entry?

**Comment by jameskoster** [comment]
Thanks for the review @tyxla :)

I'd appreciate if someone from @WordPress/gutenberg-design could take a look when they get a second.

File: packages/components/src/alignment-matrix-control/styles/alignment-matrix-control-styles.ts (hunk 5 above)
Comment (tyxla) [reply packages/components/src/alignment-matrix-control/styles/alignment-matrix-control-styles.ts:18]: I don't mind keeping it if that's what you prefer 👍

**Comment by jasmussen** [comment]
We can try it!

File: packages/components/CHANGELOG.md [hunk 8]
```diff
@@ -34,6 +34,32 @@
...
+-   `ProgressBar`: Adopt radius scale ([#64368](https://github.com/WordPress/gutenberg/pull/64368)).
+-   `RadioControl`: Adopt radius scale ([#64368](https://github.com/WordPress/gutenberg/pull/64368)).
+-   `Snackbar`: Adopt radius scale ([#64368](https://github.com/WordPress/gutenberg/pull/64368)).
+-   `TabPanel`: Adopt radius scale ([#64368](https://github.com/WordPress/gutenberg/pull/64368)).
+-   `Text`: Adopt radius scale ([#64368](https://github.com/WordPress/gutenberg/pull/64368)).
+-   `ToggleGroupControl`: Adopt radius scale ([#64368](https://github.com/WordPress/gutenberg/pull/64368)).
+-   `ToolbarGroup`: Adopt radius scale ([#64368](https://github.com/WordPress/gutenberg/pull/64368)).
+-   `Toolbar`: Adopt radius scale ([#64368](https://github.com/WordPress/gutenberg/pull/64368)).
+-   `Tooltip`: Adopt radius scale ([#64368](https://github.com/WordPress/gutenberg/pull/64368)).
+-   `UnitControl`: Adopt radius scale ([#64368](https://github.com/WordPress/gutenberg/pull/64368)).
```
Comment (ciampo) [inline packages/components/CHANGELOG.md:55]: A new release of the package happened in the meantime, and therefore we should move these items under the "Unlereased" section above.

**Review by ciampo — COMMENTED** [review]
Thank you for working on this, @jameskoster !

I think we should be able to remove the [`radiusBlockUi` variable](https://github.com/WordPress/gutenberg/blob/52171c992fbb9a43e69a25054e44de595ac7390d/packages/components/src/utils/config-values.js#L51) from the internal component config !


I also see snapshot tests failing — while we update them, let's take it as one additional way to make sure that there are no unintended changes in this PR.

**Review by ciampo — APPROVED** [review]
Changes are looking good!


We also have a few usages left of the `controlBorderRadius` shared variable (in `PaletteEdit` and `ItemGroup`) — feel free to address them here or in a follow-up PR. Either way, we should then be able to remove `controlBorderRadius` too from the shared config!

**Comment by ciampo** [comment]
@jameskoster Looking good! We can go ahead and merge whenever you're ready to :)
